SpokeStack/SpokeStack/WakeWordSpeechRecognizer.swift

```diff
@@ -0,0 +1,660 @@
+                    
+                } catch {
+                  
+                    fatalError("couldn't write the smooth window")
+                }
+                
+                indexIncrement += 1
+
+            } while indexIncrement < resultCount
+            
+        } catch let modelDetectError {
+            
+            fatalError("modelDetectError is thrown \(modelDetectError)")
+        }
+        
+        self.smooth()
+    }
+}
+
+extension WakeWordSpeechRecognizer: AudioEngineControllerDelegate {
+    
+    func didReceive(_ buffer: AVAudioPCMBuffer) {
+        self.process(buffer)
+    }
+    
+    func didStart(_ engineController: AudioEngineController) {
+        print("it did start")
+    }
+    
+    func didStop(_ engineController: AudioEngineController) {
+        
+        /// "close"
+        
+        /// Reset
+        
+        self.reset()
+    }
+}
+
+extension WakeWordSpeechRecognizer {
+    
+    private func smooth() -> Void {
+        
+        /// Sum the per-class posteriors across the smoothing window
+        
+        for (index, _) in self.words.enumerated() {
+            self.phraseSum[index] = 0
+        }
+
+        while !self.smoothWindow.isEmpty {
+            
+            for (index, _) in self.words.enumerated() {
+                
+                do {
+
+                    self.phraseSum[index] += try self.smoothWindow.read()
+
+                } catch SpeechPipelineError.illegalState(let message) {
+                    
+                    fatalError("Error occured while smoothing \(message)")
+                    
+                } catch {
+                    
+                     fatalError("Error occured while smoothing \(#line)")
+                }
+            }
+        }
+        
+        /// Compute the posterior mean of each keyword class
+        /// Write the outputs to the phrasing window
+        
+        let total: Int = self.smoothWindow.capacity / self.words.count
+        self.phraseWindow.rewind().seek(self.words.count)
+
+        for (index, _) in self.words.enumerated() {
+            
+            do {
+                
+                let windowValue: Float = self.phraseSum[index] / Float(total)
+                try self.phraseWindow.write(windowValue)
+                
+                print("windowValue \(windowValue)")
+
+            } catch SpeechPipelineError.illegalState(let message) {
+                
+                fatalError("illegal state error \(message)")
+                
+            } catch {
+                
+                fatalError("Error occured while smoothing \(#line)")
+            }
+        }
+        
+        self.phrase()
+    }
+    
+    private func phrase() -> Void {
+        
+        /// Compute the argmax (winning class) of each smoothed output
+        /// in the current phrase window
+        
+        var index: Int = 0
+        var max: Float = -Float.greatestFiniteMagnitude
```

Comment:
This max needs to be initialized within the phrase window loop, because we want to calculate the winning word class across for each phrase frame. See https://github.com/pylon/spokestack-android/blob/master/src/main/java/com/pylon/spokestack/wakeword/WakewordTrigger.java#L501.